wpiutil/src/main/java/edu/wpi/first/util/RuntimeLoader.java

```diff
@@ -4,75 +4,28 @@
 
 package edu.wpi.first.util;
 
-import java.io.File;
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.security.DigestInputStream;
-import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
-import java.util.Locale;
-import java.util.Scanner;
-
-/**
- * Loads a native library at runtime.
- *
- * @param <T> The class to load.
- */
-public final class RuntimeLoader<T> {
-  private static String defaultExtractionRoot;
-
-  /**
-   * Gets the default extraction root location (~/.wpilib/nativecache).
-   *
-   * @return The default extraction root location.
-   */
-  public static synchronized String getDefaultExtractionRoot() {
-    if (defaultExtractionRoot != null) {
-      return defaultExtractionRoot;
-    }
-    String home = System.getProperty("user.home");
-    defaultExtractionRoot = Paths.get(home, ".wpilib", "nativecache").toString();
-    return defaultExtractionRoot;
-  }
-
-  private final String m_libraryName;
-  private final Class<T> m_loadClass;
-  private final String m_extractionRoot;
-
-  /**
-   * Creates a new library loader.
-   *
-   * @param libraryName Name of library to load.
-   * @param extractionRoot Location from which to load the library.
-   * @param cls Class whose classpath the given library belongs.
-   */
-  public RuntimeLoader(String libraryName, String extractionRoot, Class<T> cls) {
-    m_libraryName = libraryName;
-    m_loadClass = cls;
-    m_extractionRoot = extractionRoot;
-  }
 
+/** Loads a native library at runtime. */
+public final class RuntimeLoader {
   /**
    * Returns a load error message given the information in the provided UnsatisfiedLinkError.
    *
+   * @param libraryName the name of the library that failed to load.
    * @param ule UnsatisfiedLinkError object.
    * @return A load error message.
    */
-  private String getLoadErrorMessage(UnsatisfiedLinkError ule) {
+  private static String getLoadErrorMessage(String libraryName, UnsatisfiedLinkError ule) {
     StringBuilder msg = new StringBuilder(512);
-    msg.append(m_libraryName)
+    msg.append(libraryName)
         .append(
             " could not be loaded from path or an embedded resource.\n"
```

Comment:
The embedded resource parts of this error message are no longer needed, since there is no case this error message will ever be caused by an embedded resource missing.